cpp/src/arrow/csv/options.h

```diff
@@ -40,6 +40,8 @@ struct ARROW_EXPORT ParseOptions {
   bool escaping = false;
   // Escaping character (if `escaping` is true)
   char escape_char = '\\';
+  // Whether values are allowed to contain CR (0x0d) and LF (0x0a) characters
+  bool newlines_in_values = false;
```

Comment:
I was initially thinking the default for this should be "true", but I would wager > 90% of CSV files do not have embedded newlines and so when someone has such a "special" file they can opt in to this additional logic

Reply:
We can change the default anyway, I just thought it would be nice to have nice numbers out of the box :-)